contracts/p0/Furnace.sol

```diff
@@ -50,6 +50,7 @@ contract FurnaceP0 is ComponentP0, IFurnace {
     /// Ratio setting
     /// @custom:governance
     function setRatio(uint192 ratio_) public governance {
+        melt();
```

Comment:
Governance may want to change the ratio while paused or frozen.  Currently the `melt()` call will revert. I think we want a try-catch here.

Reply:
good catch